compiler/rustc_trait_selection/src/traits/const_evaluatable.rs

```diff
@@ -31,7 +32,9 @@ pub fn is_const_evaluatable<'cx, 'tcx>(
 ) -> Result<(), ErrorHandled> {
     debug!("is_const_evaluatable({:?}, {:?})", def, substs);
     if infcx.tcx.features().const_evaluatable_checked {
-        if let Some(ct) = AbstractConst::new(infcx.tcx, def, substs) {
+        if let Some(ct) =
+            AbstractConst::new(infcx.tcx, def, substs).map_err(ErrorHandled::Reported)?
```

Comment:
Maybe we should just implement `From<ErrorReported> for ErrorHandled`?

Reply:
yeah